### ISSUE TITLE
OAI Functions Agent Compatibility with Anthropic Functions

### ISSUE THREAD

**Issue by dawsontoth** [langchain-ai/langchainjs#2907]
Anthropic Functions partially works with the `structured-chat-zero-shot-react-description` agent executor, but gets blocked by parsing errors. Implementing OAI functions agent compatibility would improve things pretty drastically, I hear!

**Comment by jacoblee93** [comment]
@efriis you up for this one?

**Comment by mbednarzAWS** [comment]
@efriis @jacoblee93 Happy to touch base and discuss how we can collaborate to accelerate this effort.

**Comment by efriis** [comment]
Hey @mbednarzAWS ! Just continuing the conversation here.

I started looking at this last week but haven't had a chance to get it into a usable state.

The general plan is to add an equivalent OpenAIFunctionsAgentOutputParser to python, such that you can construct a Functions agent using Anthropic in the same way you would in python, using LCEL

https://python.langchain.com/docs/modules/agents/agent_types/openai_functions_agent#using-lcel

Would you be interested in owning this implementation?

**Comment by metaskills** [comment]
> The general plan is to add an equivalent OpenAIFunctionsAgentOutputParser to python
Hey folks, new here. Thanks for the work. Just checking, did you mean to say "to JS"?

**Comment by metaskills** [comment]
Also, is this work already done?

- https://github.com/langchain-ai/langchainjs/pull/2892

**Comment by efriis** [comment]
Yes and yes!

**Comment by dawsontoth** [comment]
I believe there's some more underlying work being sorted out, the agent executor isn't quite straightened out on bedrock yet.

**Comment by efriis** [comment]
Ah I recall now - still need to implement the functions parser separately, so it plays nicely

**Comment by efriis** [comment]
If anyone wants to take a stab at it, it would essentially be a similar output parser to this python implementation (but in js)! https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/output_parsers/openai_functions.py
